src/quantity_api/__init__.py

```diff
@@ -17,5 +17,26 @@ def value(self) -> V: ...
     def unit(self) -> U: ...
 
     ### Dunder Methods
-
     def __eq__[B](self, other: Self[op.CanEq[V, B], U], /) -> B: ...
+    def __ne__[B](self, other: Self[op.CanNe[V, B], U], /) -> B: ...
+    def __lt__[B](self, other: Self[op.CanLt[V, B], U], /) -> B: ...
+    def __le__[B](self, other: Self[op.CanLe[V, B], U], /) -> B: ...
+    def __gt__[B](self, other: Self[op.CanGt[V, B], U], /) -> B: ...
+    def __ge__[B](self, other: Self[op.CanGe[V, B], U], /) -> B: ...
+    
+    def __pos__(self) -> Self[op.CanPos[V], U]: ...
+    def __neg__(self) -> Self[op.CanNeg[V], U]: ...
+    def __abs__(self) -> Self[op.CanAbs[V], U]: ...
+
+    def __add__[B](self, other: Self[op.CanAdd[V, B], U], /) -> Self[op.CanAdd[V, B], U]: ...
+    def __radd__[B](self, other: Self[op.CanAdd[V, B], U], /) -> Self[op.CanAdd[V, B], U]: ...
```

Comment:
@jorenham is this the right way to think about the reflected operators? Just want a second for this motion 😆

Reply:
The only situation where `__radd__` would be called at runtime (apart from calling it directly), is if the right-hand-side is a strict subclass of `Quantity` (it's a weird exception). So that way, we can "reflect it back" to the `__add__` of the left-hand-side, which tells us that the `other` in `__add__` should match on `op.CanRAdd` instead of `op.CanAdd`. It'll be more symmetrical that way, so it's also prettier that way :P

Reply:
And the casual higher-kinded typing notation using `Self` deserves full marks for creativity 💯

Reply:
Hmm now that I think about this a bit more, I can imagine that type-checkers could end up in an infinite loop when analyzing this, just like my brain currently is, judging by the temperature of my office.

Reply:
Oh I only just now noticed: `op.CanAdd[V, B]` describes a type with a `__add__: (self, other: V, /) -> B`. So the return type should instead be `Self[B, U]`